dashbuilder-client/dashbuilder-common-client/src/main/java/org/dashbuilder/common/client/editor/list/DropDownEditor.java

```diff
@@ -0,0 +1,210 @@
+/**
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.dashbuilder.common.client.editor.list;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.Dependent;
+import javax.enterprise.event.Event;
+import javax.inject.Inject;
+
+import com.google.gwt.editor.client.EditorError;
+import com.google.gwt.safehtml.shared.SafeHtml;
+import com.google.gwt.safehtml.shared.SafeHtmlBuilder;
+import com.google.gwt.user.client.ui.IsWidget;
+import com.google.gwt.user.client.ui.Widget;
+import org.dashbuilder.common.client.editor.LeafAttributeEditor;
+import org.dashbuilder.common.client.event.ValueChangeEvent;
+import org.dashbuilder.dataset.def.SQLDataSourceDef;
+import org.gwtbootstrap3.client.ui.constants.Placement;
+import org.uberfire.client.mvp.UberView;
+import org.uberfire.ext.widgets.common.client.dropdown.LiveSearchCallback;
+import org.uberfire.ext.widgets.common.client.dropdown.LiveSearchDropDown;
+import org.uberfire.ext.widgets.common.client.dropdown.LiveSearchService;
+
+@Dependent
+public class DropDownEditor implements IsWidget, LeafAttributeEditor<String> {
```

Comment:
Why not using generics as well on the DropDownEditor as well? Something like:

public class DropDownEditor implements IsWidget, LeafAttributeEditor

So further usages of this component with other types rather than String will be possible with no changes...